src/lib/commands.js

```diff
@@ -0,0 +1,271 @@
+import { TCModel, GVL } from "@iabtcf/core";
+import config from "./config";
+import {
+	encodeVendorConsentData,
+	encodeConsentData,
+} from "./cookie/cookie";
+const arrayFrom = require('core-js/library/fn/array/from');
+
+const COOKIE_VERSION = 2;
+const CMP_ID = 280;
+const CMP_VERSION = 2;
+
+const createCommands = (store) => {
+	let vendorList;
+
+	const getVendorList = (callback) => {
+		if (vendorList) {
+			callback(vendorList);
+		} else if (store.vendorList && Object.keys(store.vendorList).length) {
+			vendorList = JSON.parse(JSON.stringify(store.vendorList));
+			callback(vendorList);
+		} else {
+			const { getVendorList } = config;
+
+			if (getVendorList) {
+				getVendorList((err, data) => {
+					if (err) {
+						callback({}, err);
+					} else {
+						vendorList = data;
+						callback(vendorList);
+					}
+				});
+			}
+		}
+	};
+
+	const vectorToObject = (set) => {
+		let obj = {};
+
+		set.forEach((hasConsent, id) => {
+			obj[id] = hasConsent;
+		});
+
+		return obj;
+	};
+
+	 /**
+	 * Build consent fields object v1 from data that has already been persisted.
+	 */
+	const getConsentFieldsObject = () => {
+		const {
+			persistedConsentData
+		} = store;
+
+		const {
+			created = new Date(),
+			lastUpdated = new Date(),
+			cmpId_: cmpId = store.tcModel.cmpId,
+			cmpVersion_: cmpVersion = 1,
+			consentScreen_: consentScreen = store.tcModel.consentScreen,
+			consentLanguage_: consentLanguage = store.tcModel.consentLanguage,
+			vendorListVersion_: vendorListVersion,
+		} = persistedConsentData;
+
+		return {
+			cmpId,
+			cmpVersion,
+			consentLanguage,
+			consentScreen,
+			created,
+			vendorListVersion,
+			lastUpdated,
+		};
+	};
+
+	const generateConsentStringV1 = (data = {}, callback) => {
+
+		getVendorList((vendorList, err) => {
```

Comment:
You don't use `vendorList` inside callback. Is there any reason you that call `getVendorList` function here?

Reply:
Good point. Probably I thought that I will need and forgot to delete it - already deleted.